A lint that finds nothing fails the release: golangci-lint times out on erun-backend-api and the gate reports it as a failure

## What happened

A release aborts on a lint that found nothing. `make check` (run inside the `erun-devops` image build, so it gates every release) failed like this:

```
>> golangci-lint erun-backend/erun-backend-api
0 issues.
level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
>> golangci-lint erun-ui
0 issues.
lint failed in: erun-backend/erun-backend-api
make[1]: *** [Makefile:50: lint] Error 1
make: *** [Makefile:181: check] Error 2
ERROR: failed to solve: process "/bin/sh -c make check && touch /test-ok" did not complete successfully: exit code: 2
```

**`0 issues`, then a non-zero exit.** The analysis completed and was clean; golangci-lint exceeded its own default timeout and failed anyway. Nothing was wrong with the code, and the release died 22 minutes in.

## Why that module

`LINT_MODULES` is `erun-common erun-cli erun-mcp erun-integration erun-backend/erun-backend-api erun-ui`. Four of those carry their own `.golangci.yml` with an explicit `run.timeout: 3m`. **`erun-backend/erun-backend-api` and `erun-integration` have none**, so they run on the default — and `erun-backend-api` is the largest module in the repo, pulling the AWS SDK (ec2, iam, ssm, sso, signin). On a 2-core build host it sits right at the edge, so this fails intermittently rather than reproducibly, which is the worst shape for a release gate.

It also fails *silently misleadingly*: the module that timed out is named in `lint failed in:` several lines after its own `0 issues.`, so the obvious reading is that a later module found something.

## What you expected

A lint that finds nothing does not fail the build. Either the invocation carries an explicit bound, or every module gets a `.golangci.yml` with one.

I have taken the first, since it is one line and changes no linter behaviour anywhere:

```make
LINT_TIMEOUT ?= 15m
...
if ! (cd $$m && golangci-lint run --timeout $(LINT_TIMEOUT) ./...); then
```

A per-module config for `erun-backend-api` would arguably be tidier, but adding one means choosing its `linters.enable` set, which would newly apply `funlen`/`gocyclo`/`cyclop` to that module and surface a batch of unrelated findings — a separate decision, not a timeout fix.

Worth considering alongside: a timeout that is hit should probably not be reported through the same channel as a finding at all, since "the analysis could not finish" and "the analysis found a problem" need different responses from whoever reads the log.

## Reproduction

1. A build host with modest CPU (observed: 2 cores, the `erun/main` runtime pod).
2. `make check`, or any `erun release` (the `erun-devops` image runs it as a gate).
3. `erun-backend/erun-backend-api` prints `0 issues.` followed by `Timeout exceeded`, and the gate fails.

## Environment

- erun version: 1.0.211 released from this host; failure hit while cutting 1.0.212
- context: host-side orchestrator (`erun-devops`), `erun/main` runtime pod (2 CPU / 4 GiB)

--- a/Makefile
+++ b/Makefile
@@ -33,6 +33,15 @@
 #
 # erun-backend-db has no Go module at all (Atlas migrations + SQL only), so
 # there is nothing here for golangci-lint to run against.
+# An outer bound on a single module's lint, passed explicitly because the
+# default is short enough that the largest module (erun-backend-api, which
+# carries the AWS SDK) exceeds it on a modest build host -- and golangci-lint
+# reports that as a failure *after* printing "0 issues", so a clean analysis
+# reads as a red gate and a release aborts for no finding at all. Modules with
+# their own .golangci.yml set a shorter run.timeout; this is the ceiling, not a
+# replacement for it.
+LINT_TIMEOUT ?= 15m
+
 LINT_MODULES := erun-common erun-cli erun-mcp erun-integration erun-backend/erun-backend-api erun-ui
 
 # Run golangci-lint across the gated modules, each against its own
@@ -58,7 +67,7 @@ lint:
 	@failed=""; \
 	for m in $(LINT_MODULES); do \
 		echo ">> golangci-lint $$m"; \
-		if ! (cd $$m && golangci-lint run ./...); then \
+		if ! (cd $$m && golangci-lint run --timeout $(LINT_TIMEOUT) ./...); then \
 			failed="$$failed $$m"; \
 		fi; \
 	done; \

